htdocs/comm/propal/class/propal.class.php

```diff
@@ -2066,7 +2066,7 @@ public function set_date($user, $date, $notrigger = 0)
 
 			$this->db->begin();
 
-			$sql = "UPDATE ".MAIN_DB_PREFIX."propal SET datep = '".$this->db->idate($date)."'";
+			$sql = "UPDATE ".MAIN_DB_PREFIX."propal SET datep = '".$this->db->idate($date)."',fin_validite = '".$this->db->idate($this->fin_validite)."'";
```

Comment:
In all other set_date method, only the main date of object is modified.
So i suggest to not modify the "fin_validite" date here but move this inside caller, just after the ->set_date by doing a complementary update.

Reply:
@eldy OK i have changed